src/periphery/LiquidityAmounts.sol

```diff
@@ -69,39 +69,46 @@ library LiquidityAmounts {
 
     /// @notice Return the ids and liquidities of an user
     /// @param user The address of the user
+    /// @param ids the list of ids where the user have liquidity, ids need to be in ascending order to assert uniqueness
     /// @param LBPair The address of the LBPair
-    /// @return ids the list of ids where the user has liquidity
     /// @return liquidities the list of amount of liquidity of the user
-    function getLiquiditiesOf(address user, address LBPair)
-        internal
-        view
-        returns (uint24[] memory ids, uint256[] memory liquidities)
-    {
+    function getLiquiditiesOf(
+        address user,
+        uint24[] calldata ids,
+        address LBPair
+    ) internal view returns (uint256[] memory liquidities) {
         uint256 positionNumber = ILBToken(LBPair).userPositionNumber(user);
 
         liquidities = new uint256[](positionNumber);
-        ids = new uint24[](positionNumber);
 
+        uint24 id;
         for (uint256 i; i < positionNumber; ++i) {
-            uint24 id = uint24(ILBToken(LBPair).userPositionAtIndex(user, i));
+            if (id >= ids[i] && id != 0) revert LiquidityAmounts__OnlyStrictlyIncreasingId();
+            id = ids[i];
 
-            ids[i] = id;
             liquidities[i] = ILBToken(LBPair).balanceOf(user, id);
         }
     }
 
     /// @notice Return the amounts received by an user if he were to burn all its liquidity
     /// @param user The address of the user
+    /// @param ids the list of ids where the user would remove its liquidity, ids need to be in ascending order to assert uniqueness
     /// @param LBPair The address of the LBPair
     /// @return amountX the amount of tokenX received by the user
     /// @return amountY the amount of tokenY received by the user
-    function getAmountsOf(address user, address LBPair) internal view returns (uint256 amountX, uint256 amountY) {
+    function getAmountsOf(
+        address user,
+        uint24[] calldata ids,
+        address LBPair
+    ) internal view returns (uint256 amountX, uint256 amountY) {
         uint256 positionNumber = ILBToken(LBPair).userPositionNumber(user);
 
+        uint24 id;
         for (uint256 i; i < positionNumber; ++i) {
-            uint24 id = uint24(ILBToken(LBPair).userPositionAtIndex(user, i));
-            uint256 liquidity = ILBToken(LBPair).balanceOf(user, id);
+            if (id >= ids[i] && id != 0) revert LiquidityAmounts__OnlyStrictlyIncreasingId();
```

Comment:
If `id >= ids[i]` there is very low chances that `id == 0` right ?

Reply:
It is only right if `ids[0] == 0`. The id 0 shouldn't be queried as the liquidity in bin 0 will always be 0 as the power would revert. But as this check doesn't add any gas cost (I think 12wei?), we should keep it in case this function is ever copied somewhere else where this check would be necessary (also for good code).